emr-logservice/src/main/scala/org/apache/spark/sql/aliyun/logservice/LoghubSourceRDD.scala

```diff
@@ -154,41 +164,39 @@ class LoghubSourceRDD(
                   val columnArray = Array.tabulate(schemaFieldNames.length)(_ =>
                     (null, null).asInstanceOf[(String, String)]
                   )
-                  log.getContentsList
-                    .filter(content => schemaFieldPos.contains(content.getKey))
-                    .foreach(content => {
-                      columnArray(schemaFieldPos(content.getKey)) = (content.getKey, content.getValue)
-                    })
-
-                  val flg = group.GetFastLogGroup()
-                  for (i <- 0 until flg.getLogTagsCount) {
-                    val tagKey = flg.getLogTags(i).getKey
-                    val tagValue = flg.getLogTags(i).getValue
+                  for (i <- 0 until fastLogGroup.getLogTagsCount) {
+                    val tag = fastLogGroup.getLogTags(i)
+                    val tagKey = tag.getKey
+                    val tagValue = tag.getValue
+                    if (schemaFieldPos.contains(tagKey)) {
+                      columnArray(schemaFieldPos(tagKey)) = (tagKey, tagValue)
+                    }
+                  }
+                  for (i <- 0 until fastLogGroup.getLogTagsCount) {
+                    val tagKey = fastLogGroup.getLogTags(i).getKey
+                    val tagValue = fastLogGroup.getLogTags(i).getValue
                     if (schemaFieldPos.contains(s"__tag__:$tagKey")) {
                       columnArray(schemaFieldPos(s"__tag__:$tagKey")) = (s"__tag__:$tagKey", tagValue)
                     }
                   }
-
+                  if (appendSequenceNumber && schemaFieldPos.contains(__SEQUENCE_NUMBER__)) {
```

Comment:
The `appendSequenceNumber ` config is not necessary.  Checking schema if contains `__SEQUENCE_NUMBER__ ` is enough.